elastic/kibana
# [Lens] Do not refresh session on "now" drift on incoming data

Fixes https://github.com/elastic/kibana/issues/124219

This PR does not automatically start a new session if "now" drifted too much on incoming data to avoid endlessly fetching data if requests take a while to complete.

It also adds a new debug hook to test better with slow requests:
* Set `data.search.aggs.shardDelay.enabled` to true in the yml file
* Go to Lens
* Set `window.ELASTIC_LENS_DELAY_SECONDS` to a number of seconds to delay all requests by

To test this PR:
* Go to Lens and use a small dynamic time range (last 5 minutes)
* Set request delay to 20 seconds so "now" drift threshold is met before the request finishes
* Configure a chart
* Without this PR, it will load the chart, then kick off a new request and never stop loading - with this PR it should just load once, then stop
